core/trino-main/src/test/java/io/trino/sql/query/TestSubqueries.java

```diff
@@ -597,4 +597,110 @@ public void testCorrelatedJoin()
 
         assertions.assertQueryReturnsEmptyResult("SELECT * FROM (SELECT 1 where 0 = 1) t(a) LEFT JOIN LATERAL (SELECT 2 WHERE a = 1 ) t2(b) ON TRUE");
     }
+
+    @Test
+    public void testCorrelatedUnnest()
```

Comment:
Add tests with multiple unnest (e.g., `FROM UNNEST(a), UNNEST(b)`, and tests with correlations at multiple levels (i.e., two or more levels of nesting)

Reply:
> FROM UNNEST(a), UNNEST(b)

In the subquery? No, that's not supported. This rule supports single correlated UNNEST. The mechanism of this rule is that it identifies a correlated subquery based on UNNEST, and "plants" the subquery (with all necessary modifications) on top of CorrelatedJoin's input. If we tried it with two UNNESTs, we'd have to duplicate the CorrelatedJoin's input plan, decorrelate UNNESTs and join.

> tests with correlations at multiple levels (i.e., two or more levels of nesting)
There are tests like
`"SELECT (SELECT max(count) FROM (SELECT count(v) AS count FROM UNNEST(id, val) u(i, v) GROUP BY i)) "`
where correlation comes from a further outer scope.
I'll try to add some example where the innermost subquery is resolved and it enables decorrelation on another level.

Reply:
I added a test with two levels of decorrelation.